ParsingNotes.md

```diff
@@ -0,0 +1,167 @@
+`ordgroup`, but once `\mathord`, `\mathop`, and friends are implemented there
+will be a few more.  The structure of an `ordgroup` is:
+
+```
+type: "ordgroup"
+value: Array(ParseNode)
+mode: "math"
+```
+
+Functions in `functions.js` will _always_ get this kind of node when the parameters
+of the argument are enclosed in brackets `{` and `}`.  For instance, if you call
+`\sqrt 2`, then the function `\sqrt` will get a single _symbol_ `ParseNode` as defined
+in (1) containing 2.  In contrast if you call `\sqrt{2}`, then `\sqrt` will be passed
+a single _ordgroup_ `ParseNode` with a value being a list containing a single
+symbol parse node `[ ParseNode("mathord", 2) ]`.  And of course if you called 
+`\sqrt{2+1}` then you would also get an _ordgroup_ `ParseNode` with a value of a list
+containing three `ParseNode`s for each symbol in the arguments.
+
+### Functions
+
+Functions for our purposes are _most_ things that can be found in `functions.js`.
+These _usually_ aren't the kind of tokens that you would see in a reference
+implementation of TeX, but are more of a hard-coded shortcut of _most_ of the
+commands that people find useful while using TeX.  Perhaps one of the easiest
+examples is the `\sqrt` function.  The `\sqrt` function will have a `ParseNode`
+that looks like this: 
+
+```
+type: "\\sqrt"
+value:
+    type: "\\sqrt"
+    body: PaseNode("ordgroup")
+mode: "math"
+```
+
+We have already discussed in the previous section why we will _usually_ expect a
+`ParseNode("ordgroup")` in the value, but we also know that is not always the case.
+It could simply be a symbol `ParseNode`.  But functions to use their arguments to
+construct a richer structure which could be useful for processing later one.
+On exmaple comes from the generalized fractions: `\frac`, `\dfrac`, `\binom`, etc.
+These usually generate a structure that looks like: 
+
+```
+type: "genfrac"
+value:
+    type: "genfrac"
+    numer: 
+        type: "ordgroup"        // Always wrapped in single ordgroup
+        value: Array(ParseNode)
+        mode: "math"
+    denom:
+        type: "ordgroup"        // Always wrapped in single ordgroup
+        value: Array(ParseNode)
+        mode: "math"
+    hasBarLine: <Bool>
+    leftDelim: "("
+    rightDelim: ")"
+    size: "display"/"auto"/"text"
+mode: "math"
+```
+In contrast to most functions, these will _always_ have `ordgroup`s in their numer
+and denom, but this has more to do with how infix operators are parsed in general
+than with whether or not we use braces to enclose our arguments.
+
+One thing to keep in mind with functions is that they may have additional information
+as shown in the `genfrac` type above that is expected from the parser.  One last
+example, which is quite common, is the `subsup` node:
+
+```
+type: "supsub"
+value:
+    base: ParseNode  //either a symbol or ordgroup
+    sup:  ParseNode  //either a symbol or ordgroup
+    sub:  ParseNode  //either a symbol or ordgroup
+mode: "math"
```

Comment:
Essentially because it's not in `functions.js`.  Everything in `functions.js` will have a second level of indirection.  So perhaps wasn't a good example.